lib/autoproj/configuration.rb

```diff
@@ -177,7 +184,10 @@ def configure(option_name)
                 if current_value = config[option_name]
                     current_value = current_value.first
                 end
-                value = opt.ask(current_value)
+                value = opt.ask(current_value, nil, interactive: interactive?)
```

Comment:
It's weird to **ask** anything if it's not meant to actually **be asked**.

Why not leave `#ask` alone, and simply return the default value, not even storing it in the config hash ? That would remove the need for the `@use_default` variable. You would however need to take this into account in `#to_hash` (filling the defaults from options that haven't been asked yet)

Reply:
Just followed your initial suggestion, but it should be less 'weird' now